causalml/inference/meta/slearner.py

```diff
@@ -223,3 +198,131 @@ def estimate_ate(self, X, treatment, y):
         te_lb = self.model.conf_ints[0, 0]
         te_ub = self.model.conf_ints[0, 1]
         return te, te_lb, te_ub
+
+class BaseSClassifier(object):
+    """A parent class for S-learner classifier classes.
+    An S-learner estimates treatment effects with one machine learning model.
+    Details of S-learner are available at Kunzel et al. (2018) (https://arxiv.org/abs/1706.03461).
+    """
+
+    def __init__(self, learner=None, ate_alpha=0.05, control_name=0):
+        """Initialize an S-learner.
+        Args:
+            learner (optional): a model to estimate the treatment effect
+            control_name (str or int, optional): name of control group
+        """
+        if learner:
+            self.model = learner
+        else:
+            self.model = DummyRegressor()
+        self.ate_alpha = ate_alpha
+        self.control_name = control_name
+
+    def __repr__(self):
+        return '{}(model={})'.format(self.__class__.__name__,
+                                     self.model.__repr__())
+
+    def fit(self, X, treatment, y):
+        """Fit the inference model
+        Args:
+            X (np.matrix): a feature matrix
+            treatment (np.array): a treatment vector
+            y (np.array): an outcome vector
+        """
+        is_treatment = treatment != self.control_name
+        w = is_treatment.astype(int)
+
+        t_groups = np.unique(treatment[is_treatment])
+        self._classes = {}
+
+        # this should be updated for multi-treatment case
+        self._classes[t_groups[0]] = 0
+        X = np.hstack((w.reshape((-1, 1)), X))
+        self.model.fit(X, y)
+
+    def predict(self, X, treatment, y=None):
+        """Predict treatment effects.
+        Args:
+            X (np.matrix): a feature matrix
+            treatment (np.array): a treatment vector
+            y (np.array, optional): an outcome vector
+        Returns:
+            (numpy.ndarray): Predictions of treatment effects.
+        """
+        is_treatment = treatment != self.control_name
+        w = is_treatment.astype(int)
+
+        X = np.hstack((w.reshape((-1, 1)), X))
+
+        X[:, 0] = 0
+        yhat_c = self.model.predict_proba(X)[:, 1]
+
+        X[:, 0] = 1
+        yhat_t = self.model.predict_proba(X)[:, 1]
+
+        if y is not None:
+            logger.info('RMSE (Control): {:.6f}'.format(
+                np.sqrt(mse(y[~is_treatment], yhat_c[~is_treatment])))
+            )
+            logger.info(' MAE (Control): {:.6f}'.format(
+                mae(y[~is_treatment], yhat_c[~is_treatment]))
+            )
+            logger.info('RMSE (Treatment): {:.6f}'.format(
+                np.sqrt(mse(y[is_treatment], yhat_t[is_treatment])))
+            )
+            logger.info(' MAE (Treatment): {:.6f}'.format(
+                mae(y[is_treatment], yhat_t[is_treatment]))
+            )
+
+        return (yhat_t - yhat_c).reshape(-1, 1)
```

Comment:
We might want to add np.clip here to ensure values are between 0 and 1 (e.g. if yhat_c > yhat_t)